parser: remove CREATE DATABASE ... REGION from docs

otan opened the pull request:
Release note (sql change): Remove usage of `CREATE DATABASE ... REGION =
...` from documentation. However, this is still usable.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57006)
<!-- Reviewable:end -->

otan committed:
parser: remove CREATE DATABASE ... REGION from docs

Release note (sql change): Remove usage of `CREATE DATABASE ... REGION =
...` from documentation. However, this is still usable.

ajstorm reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57006#-:-MMp2SUd2DKRyKEG6Tng:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @ajstorm)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajstorm
